### PR TITLE
Refactored PopularityCommunity

### PR DESCRIPTION
**Pull request by devos50**
In this PR, I refactored the `PopularityCommunity` and simplified it. In essence, it now gossips around at most 10 torrents. Five of these torrents are randomly selected torrents that you have checked (with at least one seeder). The other five torrents are the most healthy torrents that are health checked by you.

Addresses #4256 

Complements https://github.com/Tribler/gumby/pull/409

### CODE REVIEW

File: Tribler/community/popularity/payload.py [hunk 1]
```diff
@@ -1,48 +1,57 @@
...
     def to_pack_list(self):
-        data = [('20s', self.infohash),
-                ('I', self.num_seeders),
-                ('I', self.num_leechers),
-                ('Q', self.timestamp)]
+        random_torrents_items = [item for sublist in self.random_torrents for item in sublist]
+        checked_torrents_items = [item for sublist in self.torrents_checked for item in sublist]
+        data = [('I', len(self.random_torrents)),
+                ('I', len(self.torrents_checked)),
+                ('raw', struct.pack("!" + TORRENT_INFO_FORMAT * len(self.random_torrents), *random_torrents_items)),
```
Comment (qstokkink) [inline Tribler/community/popularity/payload.py]: This should be a `varlen` variant, `raw` can only appear once at the end of a message.

Comment (devos50) [reply Tribler/community/popularity/payload.py]: Thanks! 👍 this is a bug indeed

**Comment by devos50** [comment]
I made a validation experiment to verify the correctness of the popularity community, see https://jenkins-ci.tribler.org/job/validation_experiments/job/validation_experiment_popularity/

File: Tribler/community/popularity/community.py [hunk 2]
```diff
@@ -1,130 +1,82 @@
...
-        """
-        Method to send torrent health response. This message is sent to all the subscribers by default but if a
-        peer is specified then only that peer receives this message.
-        """
-        if peer and peer not in self.get_peers():
-            self.logger.debug(ERROR_UNKNOWN_PEER)
-            return
+        num_torrents_checked = len(self.torrent_checker.torrents_checked)
+        random_torrents_checked = random.sample(self.torrent_checker.torrents_checked, min(num_torrents_checked, 5))
+        popular_torrents_checked = sorted(self.torrent_checker.torrents_checked,
```
Comment (xoriole) [inline Tribler/community/popularity/community.py]: Possibility of duplicate torrents for a small set?

**Pull request merged**